projects/ion/src/lib/picker/control-picker/control-picker.component.spec.ts

```diff
@@ -0,0 +1,111 @@
+): Promise<void> => {
+  await render(IonControlPickerComponent, {
+    componentProperties: customProps,
+    declarations: [],
+    imports: [IonButtonModule, IonTooltipModule, IonDividerModule],
+  });
+};
+
+describe('IonControlPickerComponent', () => {
+  it('should emit an event on clicking the previous year button', async () => {
+    const onPreviousYear = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onPreviousYear });
+    const btnPreviousYear = screen.getByTestId('btn-previous-year');
+    jest.spyOn(onPreviousYear, 'emit');
+    fireEvent.click(btnPreviousYear);
+    expect(onPreviousYear.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit an event on clicking the previous month button', async () => {
+    const onPreviousMonth = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onPreviousMonth });
+    const btnPreviousMonth = screen.getByTestId('btn-previous-month');
+    jest.spyOn(onPreviousMonth, 'emit');
+    fireEvent.click(btnPreviousMonth);
+    expect(onPreviousMonth.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit an event on clicking the next month button', async () => {
+    const onNextMonth = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onNextMonth });
+    const btnNextMonth = screen.getByTestId('btn-next-month');
+    jest.spyOn(onNextMonth, 'emit');
+    fireEvent.click(btnNextMonth);
+    expect(onNextMonth.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit an event on clicking the next year button', async () => {
+    const onNextYear = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onNextYear });
+    const btnNextYear = screen.getByTestId('btn-next-year');
+    jest.spyOn(onNextYear, 'emit');
+    fireEvent.click(btnNextYear);
+    expect(onNextYear.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit an event on clicking the month button', async () => {
+    const onChangeMonth = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onChangeMonth });
+    const labelMonth = screen.getByTestId('label-month');
+    jest.spyOn(onChangeMonth, 'emit');
+    fireEvent.click(labelMonth);
+    const button = screen.getByTestId('btn-month-2');
+    fireEvent.click(button);
+    expect(onChangeMonth.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should emit an event on clicking the year button', async () => {
+    const onChangeYear = new EventEmitter<ControlEvent>();
+    await sut({ ...defaultComponent, controlPickerEvent: onChangeYear });
+    const labelYear = screen.getByTestId('label-year');
+    jest.spyOn(onChangeYear, 'emit');
+    fireEvent.click(labelYear);
+    const button = screen.getByTestId('btn-year-2025');
+    fireEvent.click(button);
+    expect(onChangeYear.emit).toHaveBeenCalledTimes(1);
+  });
```

Comment:
This test will broken next year?

Reply:
it's not for breaking because the component was set to 2022 regardless of the year component is tested.

`const defaultComponent: IonControlPickerComponentProps = {
  month: 'Janeiro',
  year: '2022', 
  ...
}`